cPP_DSC.adoc

```diff
@@ -674,28 +674,28 @@ _Based on their implementation, vendors will explain when certain keys are no lo
+... [.underline]#erasure of all logical blocks in the memory device (all physical blocks except the index blocks, if they can be reliably identified) followed by a read-verify operation on all logical blocks];#
 .. [.underline]#that does not employ a wear-leveling algorithm, the destruction shall be executed by a [*selection*:#
-... [.underline]#[*selection*: single, [*_assignment:* ST author defined multi-pass_]] overwrite consisting of [*selection*: zeros, ones, pseudo-random pattern, a new value of a key of the same size, [*_assignment:* some value that does not contain any CSP_]] followed by a read-verify. If the read-verification of the overwritten data fails, the process shall be repeated again up to [*_assignment:* number of times to attempt overwrite_] times, whereupon an error is returned.#
-... [.underline]#block erase]#
+... [.underline]#[*selection*: single, [*_assignment*: ST author defined multi-pass_]] overwrite consisting of [*selection*: zeros, ones, pseudo-random pattern, a new value of a key of the same size, [*_assignment*: some value that does not contain any CSP_]] followed by a read-verify. If the read-verification of the overwritten data fails, the process shall be repeated up to [*_assignment*: number of times to attempt overwrite_] times, whereupon an error is returned.#
+... [.underline]#block erase followed by a read-verify. If the read-verification of the overwritten data fails, the process shall be repeated up to [*_assignment*: number of times to attempt overwrite_] times, whereupon an error is returned.]#
 +
 ]] that meets the following: [_no standard_].
 
 _Application Note {counter:remark_count}_:: _In the case of volatile memory, the selection "removal of all references to the key directly followed by a request for garbage collection" is used in a situation where the TSF cannot address the specific physical memory locations holding the data to be erased and therefore relies on addressing logical addresses (which frees the relevant physical addresses holding the old data) and then requesting the platform to ensure that the data in the physical addresses is no longer available for reading (i.e. the "garbage collection" referred to in the SFR text)._
 +
-_The selection for destruction of data in non-volatile memory includes block erase as an option, and this option applies only to flash memory. A block erase does not require a read verify, since the mappings of logical addresses to the erased memory locations are erased as well as the data itself._
+_In the case of wear-leveled non-volatile memory, the entire memory must be erased when a key is deleted unless the TOE has an erasure mechanism that meets FDP_RIP.1 and the vendor provides a rationale that the TOE uses it in a conformant way.  FDP_RIP.1 requires that keys not remain in unallocated memory after a key delete operation, a capability that wear-leveled non-volatile memory normally does not provide._
```

Comment:
Again this is circular with the reference to meeting FDP_RIP.1. That SFR only states that keys will be cleared when no longer used, this is the SFR that specifies how that happens, so I don't see how you can say the "'how" SFR mechanism should refer to the "what" SFR. The point here is to meet the key clearing requirement in an equivalent manner to that specified here, so that reference does not make sense.

You are reading FDP_RIP.1 too strictly. It says the key is made unavailable. This can mean many things, but does not specify that it has to be explicitly deleted in some manner. FCS_CKM.6 is providing implementation guidance to meet FDP_RIP.1, not the other way around. To an application within the system, deleting the logical address in wear-leveled storage and only returning 0s if anyone asks to re-read that section makes it unavailable. That may not mean it is protected against physical attacks that can directly probe the storage, but FDP_RIP.1 does not specify any more than that. 

This app note would basically preclude the use of wear-leveled storage for any DSC, and this presents serious implementation problems for some of the expected systems that would be considered as acceptable based on the original scope of this cPP. A DSC that relies on storage provided by the larger device (where the DSC encrypts the data before it is stored in the external storage) would imply that this is not acceptable because the wider storage system will not be able to ensure that the key would be deleted, even though the key is stored as encrypted by the DSC (and most likely again by the OS). This may work for a eSE that is self-contained (I think, anyway), but not for ones that may be centered around a TEE which would generally also encompass an eSE as well as other components to make up a secure system for a larger device. I fear that this app note could literally restrict the possible configurations to a degree as to make the cPP useless (since anything that is likely to work otherwise is probably already going to evaluate to either PP0084 or PP0117).